### PR TITLE
Latest SDK + Code Style with Detekt + Tests

### PR DESCRIPTION
**Pull request by daniftodi**
- using the latest version of Android Mobile SDK
- added tests
- added code style check with Detekt

### CODE REVIEW

**Review by alexanderkranga — APPROVED** [review]

File: config/detekt.yml [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+    exceptionNames:
+     - Error
+     - Exception
+     - Throwable
+     - RuntimeException
+
+formatting:
+  active: true
+  android: false
+  autoCorrect: true
```
Comment (mcyster) [inline config/detekt.yml]: I guess - little suspicous of having stuff automatically change the source code

File: config/detekt.yml [hunk 2]
```diff
@@ -0,0 +1,746 @@
...
+    autoCorrect: true
+  StringTemplate:
+    active: true
+    autoCorrect: true
+
+naming:
+  active: true
+  ClassNaming:
+    active: true
+    excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**']
```
Comment (mcyster) [inline config/detekt.yml]: why the excludes? (same question below - why are test classes different?)

**Review by mcyster — APPROVED** [review]
:+1: looks good

File: config/detekt.yml (hunk 1 above)
Comment (daniftodi) [reply config/detekt.yml]: changed it to `false`

File: config/detekt.yml (hunk 2 above)
Comment (daniftodi) [reply config/detekt.yml]: I changed this, now tests will follow the same rules, they are not different.

**Pull request merged**